### ISSUE TITLE
With Encore, Asset doesn't create the right url  (missing prefix) when using manifest.json

### ISSUE THREAD

**Issue by pfleu** [symfony/symfony#28612]
Symfony 4.1.4

In dev, using WebpackEncore, when I load my page, the browser tries to load assets (css and js generated in build) with paths where the "project_name/public" part is missing.
**Wrong path:** http://localhost/build/app.js
**Should be:** http://localhost/project_name/public/build/app.js

If I comment the content of assets.yaml, it works as expected:
```yaml
framework:
    assets:
#        json_manifest_path: '%kernel.project_dir%/public/build/manifest.json'
```

**Comment by mablae** [comment]
Your webroot is wrong. The webserver should directly point to /public for security reasons.

**Comment by javiereguiluz** [comment]
@pfleu maybe @mablae's comment was a bit strongly worded ... but he's technically right. The root dir of your web server should be configured to point to the `public/` dir of your project, not to the root dir of your project.

Please review the official doc about configuring web servers: https://symfony.com/doc/current/setup/web_server_configuration.html and if the error persists, please reopen this issue. Thanks!

**Comment by pfleu** [comment]
Thanks @mablae and @javiereguiluz for your explanations. I have been using Symfony 2 and Symfony 3 for a couple of years with the same settings and never had problems, that's why I didn't understand what happened (I wasn't using Encore though).
So I guess I should either use PHP's built-in server or use VirtualHost to change the root for each project using Encore ?

**Comment by javiereguiluz** [comment]
When using Encore, you can tell Symfony which is the public path where assets are published:

```js
// webpack.config.js
var Encore = require('@symfony/webpack-encore');

Encore
    // directory where compiled assets will be stored
    .setOutputPath('public/build/')

    // public path used by the web server to access the output path
    .setPublicPath('/project_name/public/build')

    // ...
;
```

It's explained here: https://symfony.com/doc/current/frontend/encore/simple-example.html#configuring-encore-webpack

However, even if this may work, it's strongly recommended to set `public/` as the root dir in the web server. Otherwise you may introduce security issues like this one: https://twitter.com/svblxyz/status/1045013939904532482

**Comment by i-Tor** [comment]
I don't know if I've done something wrong, but I have fixed this in a different manner.
I'm running Symfony 5.4 and following the SymfonyCasts WebpackEncore tutorial.

I have deployed my Symfony app inside a subfolder in my Apache server. So I have /var/www/subfolder/-symfony app dirs-

I have created a virtualhost in apache, so anything going to https://mydomain.com/-appname- loads the public folder of the Symfony app
I was getting errors because the encore_entry_*_tags was not properly adding the <appname> prefix.

Just adding the solution proposed by @javiereguiluz didn't work for me, as encore was telling me
`  Error: Cannot determine how to prefix the keys in manifest.json. Call Encore.setManifestKeyPrefix() to choose what path (e.g. build/) to use when building your manifest k  eys. This is caused by setOutputPath() (/var/www/<appdir>/public/build) and setPublicPath() (<appname>/build) containing paths that don't seem compatible.
`

After playing around with the options in webpack.config.js, I have managed to make it work with

```
.setPublicPath('/<appname>/build')
// only needed for CDN's or sub-directory deploy
.setManifestKeyPrefix('<appname>/') // <- Adding this is the part that worked
```

Maybe I'm doing something wrong, so far it just works when everything I tried before failed.
Thanks for the thread, helped me getting to know the ropes a bit more.
